core/src/test/java/org/elasticsearch/indices/IndexingMemoryControllerTests.java

```diff
@@ -181,7 +181,7 @@ public void testShardAdditionAndRemoval() {
         IndexService test = indicesService.indexService(resolveIndex("test"));
 
         MockController controller = new MockController(Settings.builder()
-                .put(IndexingMemoryController.INDEX_BUFFER_SIZE_SETTING, "4mb").build());
+                                                       .put(IndexingMemoryController.INDEX_BUFFER_SIZE_SETTING.getKey(), "4mb").build());
```

Comment:
I've never liked using the constants like this here in tests. The reason that I don't like is because if someone has their IDE open and accidentally inserts a character into one of the settings keys (e.g., `indices.memory.index_buffer_size` -> `indices.memory.index_buffer_sizes`) then the tests will continue to pass and so we've silently broken things. I would prefer the string literals just be copied and pasted here: `"indices.memory.index_buffer_size"` instead of `INDEX_BUFFER_SIZE_SETTING.getKey()`. Break the build early and often. :smile: